testsuite/tests/lib-unix/common/gethostbyname.ml

```diff
@@ -0,0 +1,14 @@
+(* TEST
+* hasunix
+include unix
+** bytecode
+** native
+*)
+
+(* Checks that gethostbyname supports both IPv4 and IPv6 (see #11461) *)
+let () =
+  let addr = Unix.inet_addr_of_string "127.0.0.1" in
+  ignore @@ Sys.opaque_identity Unix.gethostbyaddr addr;
+  let addr = Unix.inet_addr_of_string "::1" in
+  ignore @@ Sys.opaque_identity Unix.gethostbyaddr addr;
```

Comment:
Why `opaque_identity`?  Also, you could check parts of the result, e.g. the protocol family:
```suggestion
let check a ty =
  let addr = Unix.inet_addr_of_string a in
  let host = Unix.gethostbyaddr addr in
  assert (host.Unix.h_addrtype = ty)

let () =
  check "127.0.0.1" Unix.PF_INET;
  check "::1" Unix.PF_INET6;
```

Reply:
Thank you, I integrated this suggestion.

> Why opaque_identity?

I'm always a bit unsure about what might be optimized away.